build_projects/dotnet-cli-build/CompileTargets.cs

```diff
@@ -177,26 +179,56 @@ private static void RemovePdbsFromDir(string directory)
 
             foreach (var project in ProjectsToPublish)
             {
-                dotnet.Publish(
-                    "--native-subdirectory",
-                    "--output", sdkOutputDirectory,
-                    "--configuration", configuration,
-                    "--version-suffix", buildVersion.CommitCountString,
-                    Path.Combine(srcDir, project))
+                if (isServiceable)
+                {
+                    dotnet.Publish(
+                        "--Serviceable",
+                        "--native-subdirectory",
+                        "--output", sdkOutputDirectory,
+                        "--configuration", configuration,
+                        "--version-suffix", buildVersion.CommitCountString,
+                        Path.Combine(srcDir, project))
+                    .Execute()
+                    .EnsureSuccessful();
+                }
+                else
```

Comment:
Same applies below.

---
In reply to: [67076695](https://github.com/dotnet/cli/pull/3578#discussion_r67076695) [](ancestors = 67076695)